### PR TITLE
Python: Fix FC stepwise planner.

### LINKED ISSUE #6350: Python: FunctionCallingStepwisePlanner not working in semantic-kernel 1.0.0
**Describe the bug**
I encountered this error:
("<class 'semantic_kernel.connectors.ai.open_ai.services.azure_chat_completion.AzureChatCompletion'> service failed to complete the prompt", BadRequestError('Error code: 400 - {\'error\': {\'message\': "Invalid parameter: messages with role \'tool\' must be a response to a preceeding message with \'tool_calls\'.", \'type\': \'invalid_request_error\', \'param\': \'messages.[4].role\', \'code\': None}}'))

**To Reproduce**

**Expected behavior**

**Screenshots**

**Platform**
 - OS: Windows
 - IDE: VS Code
 - Language: Python
 - Source: semantic-kernel 1.0.0

**Additional context**

### PR DESCRIPTION
**Pull request by moonbox3**
### Motivation and Context

The FC stepwise planner was breaking when trying to process the function call result because if a filter is not configured, then the context will be None. We need to only try to extract the FC result if the context is not none, otherwise it will already be present in the chat history.

### Description

Check that the FC result context is not None, if so, then extract the FC result, otherwise proceed as the FC result is in the chat history. Fixes #6350.
- Removes some extranous logging from json schema addition.

### Contribution Checklist

- [X] The code builds clean without any errors or warnings
- [X] The PR follows the [SK Contribution Guidelines](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md) and the [pre-submission formatting script](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md#development-scripts) raises no violations
- [X] All unit tests pass, and I have added new tests where possible
- [X] I didn't break anyone :smile:

### CODE REVIEW

**Comment by markwallace-microsoft** [comment]
<a href="https://github.com/microsoft/semantic-kernel/blob/cfc794958761646c902e419411a21e1117eb8870/README.md">Py3.10 Test Coverage</a><details><summary>Python 3.10 Test Coverage Report • </summary><table><tr><th>File</th><th>Stmts</th><th>Miss</th><th>Cover</th><th>Missing</th></tr><tbody><tr><td colspan="5"><b>semantic_kernel/planners/function_calling_stepwise_planner</b></td></tr><tr><td>&nbsp; &nbsp;<a href="https://github.com/microsoft/semantic-kernel/blob/cfc794958761646c902e419411a21e1117eb8870/python/semantic_kernel/planners/function_calling_stepwise_planner/function_calling_stepwise_planner.py">function_calling_stepwise_planner.py</a></td><td>109</td><td>42</td><td>61%</td><td><a href="https://github.com/microsoft/semantic-kernel/blob/cfc794958761646c902e419411a21e1117eb8870/python/semantic_kernel/planners/function_calling_stepwise_planner/function_calling_stepwise_planner.py#L125-L212">125&ndash;212</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/cfc794958761646c902e419411a21e1117eb8870/python/semantic_kernel/planners/function_calling_stepwise_planner/function_calling_stepwise_planner.py#L246-L251">246&ndash;251</a></td></tr><tr><td colspan="5"><b>semantic_kernel/prompt_template/utils</b></td></tr><tr><td>&nbsp; &nbsp;<a href="https://github.com/microsoft/semantic-kernel/blob/cfc794958761646c902e419411a21e1117eb8870/python/semantic_kernel/prompt_template/utils/jinja2_system_helpers.py">jinja2_system_helpers.py</a></td><td>48</td><td>3</td><td>94%</td><td><a href="https://github.com/microsoft/semantic-kernel/blob/cfc794958761646c902e419411a21e1117eb8870/python/semantic_kernel/prompt_template/utils/jinja2_system_helpers.py#L15">15</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/cfc794958761646c902e419411a21e1117eb8870/python/semantic_kernel/prompt_template/utils/jinja2_system_helpers.py#L45">45</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/cfc794958761646c902e419411a21e1117eb8870/python/semantic_kernel/prompt_template/utils/jinja2_system_helpers.py#L50">50</a></td></tr><tr><td colspan="5"><b>semantic_kernel/schema</b></td></tr><tr><td>&nbsp; &nbsp;<a href="https://github.com/microsoft/semantic-kernel/blob/cfc794958761646c902e419411a21e1117eb8870/python/semantic_kernel/schema/kernel_json_schema_builder.py">kernel_json_schema_builder.py</a></td><td>45</td><td>2</td><td>96%</td><td><a href="https://github.com/microsoft/semantic-kernel/blob/cfc794958761646c902e419411a21e1117eb8870/python/semantic_kernel/schema/kernel_json_schema_builder.py#L32">32</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/cfc794958761646c902e419411a21e1117eb8870/python/semantic_kernel/schema/kernel_json_schema_builder.py#L57">57</a></td></tr><tr><td><b>TOTAL</b></td><td><b>6232</b></td><td><b>1030</b></td><td><b>83%</b></td><td>&nbsp;</td></tr></tbody></table></details>

## Python 3.10 Unit Test Overview
| Tests | Skipped | Failures | Errors | Time |
| ----- | ------- | -------- | -------- | ------------------ |
| 1341 | 1 :zzz: | 0 :x: | 0 :fire: | 13.046s :stopwatch: |

**Review by alliscode — APPROVED** [review]

File: python/samples/concepts/plugins/openai_function_calling_with_custom_plugin.py [hunk 1]
```diff
@@ -1,5 +1,6 @@
 # Copyright (c) Microsoft. All rights reserved.
 
+from __future__ import annotations
```
Comment (juliomenendez) [inline python/samples/concepts/plugins/openai_function_calling_with_custom_plugin.py:3]: I think we have another PR where this type of import is being removed.

File: python/tests/integration/planning/function_calling_stepwise_planner/test_int_function_calling_stepwise_planner.py [hunk 2]
```diff
@@ -19,7 +19,6 @@
 
 
 @pytest.mark.asyncio
-@pytest.mark.xfail(reason="This test is flaky and needs investigation.")
```
Comment (juliomenendez) [inline python/tests/integration/planning/function_calling_stepwise_planner/test_int_function_calling_stepwise_planner.py:22]: Is the test stable enough now?

**Review by juliomenendez — CHANGES_REQUESTED** [review]

File: python/samples/concepts/plugins/openai_function_calling_with_custom_plugin.py (hunk 1 above)
Comment (moonbox3) [reply python/samples/concepts/plugins/openai_function_calling_with_custom_plugin.py:3]: Yeah, I had removed it, and I don't think I pushed my change.

File: python/tests/integration/planning/function_calling_stepwise_planner/test_int_function_calling_stepwise_planner.py (hunk 2 above)
Comment (moonbox3) [reply python/tests/integration/planning/function_calling_stepwise_planner/test_int_function_calling_stepwise_planner.py:22]: yes, now that the other bug was fixed, the test passes

**Review by juliomenendez — APPROVED** [review]

**Review by juliomenendez — APPROVED** [review]

**Pull request merged**